src/main/java/org/checkerframework/specimin/unsolved/UnsolvedFieldAlternates.java

```diff
@@ -50,12 +87,40 @@ public static UnsolvedFieldAlternates create(
 
     UnsolvedFieldAlternates result = new UnsolvedFieldAlternates(alternateDeclaringTypes);
 
-    UnsolvedField field = new UnsolvedField(name, type, isStatic, isFinal);
+    UnsolvedField field = new UnsolvedField(name, type, isStatic, isFinal, Set.of());
     result.addAlternate(field);
 
     return result;
   }
 
+  /**
+   * Updates field types and must preserve nodes. Saves the intersection of the previous and the
+   * input, since we know more information to narrow potential field types down.
+   *
+   * @param typesToPreserveNodes A map of field types to nodes that must be preserved
+   */
+  public void updateFieldTypesAndMustPreserveNodes(
+      Map<MemberType, CallableDeclaration<?>> typesToPreserveNodes) {
+    // Update in-place; intersection = removing all elements in the original set
+    // that isn't found in the updated set
```

Comment:
Can we have one implementation of this intersection logic that's shared by the different `XAlternates` classes?